src/main/java/de/tum/in/www1/artemis/web/rest/QuizExerciseResource.java

```diff
@@ -215,7 +209,7 @@ public QuizExerciseResource(QuizExerciseService quizExerciseService, QuizExercis
         log.debug("REST request to get all QuizExercises for the course with id : {}", courseId);
         var course = courseRepository.findByIdElseThrow(courseId);
         if (!authCheckService.isAtLeastInstructorInCourse(course, null)) {
-            throw new AccessForbiddenException("You are not allowed to access this resource");
+            throw new AccessForbiddenException(NOT_ALLOWED);
         }
```

Comment:
Note that there are more locations where I have the same question:
- https://github.com/ls1intum/Artemis/pull/3126/files?file-filters[]=.java#diff-213fcc3ef5efb1ce4afd49700076bd8167f4cae407e4c247b3b70906e766b8a8R343-R345
- https://github.com/ls1intum/Artemis/pull/3126/files?file-filters[]=.java#diff-6c69798ea3bea86fa6d045f8bacfe798afe8f68cb2d4f71f0038eb0b969e8694R572-R589
- https://github.com/ls1intum/Artemis/pull/3126/files?file-filters[]=.java#diff-6c69798ea3bea86fa6d045f8bacfe798afe8f68cb2d4f71f0038eb0b969e8694R290-R292
- https://github.com/ls1intum/Artemis/pull/3126/files?file-filters[]=.java#diff-dd0f8f09ae82ed9969dbc331a4941ab6e942bdc0a2e622bb0532d9662ca94608R834-R836
- https://github.com/ls1intum/Artemis/pull/3126/files?file-filters[]=.java#diff-0bf9cd13c3e5e99860e2b35d8b4afe5336a1813c1ac98333cde2dbd9ccc1a129R801-R841
- https://github.com/ls1intum/Artemis/pull/3126/files?file-filters[]=.java#diff-fac115e36e233407265fdcb2d53f7aed9c07386ffd44f657e43574a0953d7eecR67-R171
- https://github.com/ls1intum/Artemis/pull/3126/files?file-filters[]=.java#diff-e523a852fd496044ee5798c0060469790be767abb3baacb046dc6f4f1741fb86R63-R65
- https://github.com/ls1intum/Artemis/pull/3126/files?file-filters[]=.java#diff-c8981cc5e7f04e7084e297989da9a545bdc3d79c630b484bdcfcbfafaf26bea2R104-R106

Reply:
I noticed now that there is a different message, so that might be the reason. Still, in some places - especially here - I don't see when we should use what method of testing the authorization.

Reply:
It's a lot of work to change all places to checkIsAtLeastInstructorInCourseElseThrow. In some cases, the logic has to be slightly changed. I just started with it to have some "good" examples and would propose to further change all other places in follow-up PRs. But feel free to help here.